common/test/acceptance/tests/lms/test_lms_courseware.py

```diff
@@ -542,3 +544,117 @@ def test_courseware_a11y(self):
+            self.browser, username=self.USERNAME, email=self.EMAIL,
+            course_id=self.course_id, staff=False
+        ).visit()
+
+        self.courseware_page.visit()
+        self.problem_page = ProblemPage(self.browser)
+
+    def get_multiple_choice_problem(self, problem_name):
+        """
+        Return the Multiple Choice Problem Descriptor, given the name of the problem.
+        """
+        factory = MultipleChoiceResponseXMLFactory()
+        xml_data = factory.build_xml(
+            question_text='The correct answer is Choice 2',
+            choices=[False, False, True, False],
+            choice_names=['choice_0', 'choice_1', 'choice_2', 'choice_3']
+        )
+
+        return XBlockFixtureDesc(
+            'problem',
+            problem_name,
+            data=xml_data,
+            metadata={'rerandomize': 'always'}
+        )
+
+    @ddt.data(
+        ('check', 'choice_choice_1', 'label.choicegroup_incorrect'),
+        ('reset', 'choice_choice_1', ['label.choicegroup_incorrect', 'span.unanswered']),
+        ('save', 'choice_choice_1', 'div.capa_alert'),
+    )
+    @ddt.unpack
+    def test_perform_problem_actions_and_navigate(self, action, selected_choice, status_selector):
+        # Scenario:
+        # I go to sequential position 1
+        # Facing problem1, I select 'choice_1'
+        # Then perform the action – check/save/reset
+        # Then I go to sequential position 2
+        # Then I came back to sequential position 1 again
+        # Facing problem1, I observe the problem1 content is not
+        # outdated before and after sequence navigation
+
+        # Go to sequential position 1 and assert that we are on problem 1.
+        self.courseware_page.go_to_sequential_position(1)
+        self.assertEquals(self.problem_page.problem_name, self.problem1_name)
+
+        # Update problem 1's content state – by performing check/reset/save operation.
+        self.problem_page.click_choice(selected_choice)
+        if action == 'check':
+            self.problem_page.click_check()
+            self.problem_page.wait_for_expected_status(status_selector, 'incorrect')
+        elif action == 'reset':
+            self.problem_page.click_check()
+            self.problem_page.wait_for_expected_status(status_selector[0], 'incorrect')
+            self.problem_page.click_reset()
+            self.problem_page.wait_for_expected_status(status_selector[1], 'unanswered')
+        elif action == 'save':
+            self.problem_page.click_save()
+            self.problem_page.wait_for_expected_status(status_selector, 'saved')
+
+        # Save the problem 1 content state as switching sequence position is about to happen.
+        problem1_content_before_switch = self.problem_page.problem_content
+
+        # Go to sequential position 2 and assert that we are on problem 2.
+        self.courseware_page.go_to_sequential_position(2)
+        self.assertEquals(self.problem_page.problem_name, self.problem2_name)
```

Comment:
nit: use `assertEqual` ([`assertEquals` is deprecated](https://docs.python.org/2/library/unittest.html#deprecated-aliases))